lib/asset/AssetGenerator.js

```diff
@@ -157,35 +185,7 @@ class AssetGenerator extends Generator {
 							);
 						}
 
-						let encodedContent;
-						if (
-							module.resourceResolveData &&
-							module.resourceResolveData.encoding === encoding
```

Comment:
I think I would also work to add a check here to check if the content is still equal: `decodeDataUriContent(module.resourceResolveData.encodedContent, module.resourceResolveData.encoding).equals(originalSource.buffer())`

So we keep encoding intact if possible.

Reply:
They must be equal when no loader transformation has happened

Reply:
@sokra Fixed